### ISSUE TITLE
NullReferenceException in OnLeave handler

### ISSUE THREAD

**Issue by ncfavier** [lemon-sh/IRCrarria#2]
I get this exception from time to time:

```
[Server API] Warning Plugin "IRCrarria" has had an unhandled exception thrown by one of its ServerLeave handlers:
System.NullReferenceException: Object reference not set to an instance of an object
  at IRCrarria.IRCrarria.OnLeave (TerrariaApi.Server.LeaveEventArgs args) [0x00021] in <a693c28872ab49038b852c39b298eea0>:0
  at TerrariaApi.Server.HandlerCollection`1[ArgsType].Invoke (ArgsType args) [0x0004d] in <4ef9d64f26f14dbd93f29814fd475187>:0
```

Nothing else in the log seems to indicate a cause.

I think this would be solved by adding some checks in `OnJoin` and `OnLeave`, like it's done [here](https://github.com/Pryaxis/TShock/blob/fa46ac65fb95d2a3a83369df4c03fb96fec48c32/TShockAPI/TShock.cs#L1257-L1267).

**Comment by lemon-sh** [comment]
Hello again!

Thanks for the interest in my plugin!
Interestingly, on my server, I found some Chinese bot IPs establishing TCP connections to the Terraria server which triggers the same bug. This would indicate that this issue happens when the Terraria handshake isn't finished successfully (automatic bots?). This bug will be fixed in the 1.2 release, once I'm done polishing the IRC framework.

Thank you so much for letting me know, have a great day <3

**Comment by lemon-sh** [comment]
Hello,
Due to personal reasons, I was unable to develop this plugin for the past couple of weeks, but today I've released the final 1.2 version with the mentioned bug fixed, if you're still interested. Sorry for the enormous delay.

**Comment by ncfavier** [comment]
Thanks!
